nucypher/cli/painting.py

```diff
@@ -173,36 +182,144 @@ def paint_known_nodes(emitter, ursula) -> None:
         emitter.echo(row_template.format(node.rest_url().ljust(20), node), color=color_index[node_type])
 
 
-def paint_contract_status(blockchain, emitter):
+def paint_contract_status(registry, emitter):
+    blockchain = BlockchainInterfaceFactory.get_interface()
 
-    token_agent = NucypherTokenAgent(blockchain=blockchain)
-    staking_agent = StakingEscrowAgent(blockchain=blockchain)
-    policy_agent = PolicyManagerAgent(blockchain=blockchain)
-    adjudicator_agent = AdjudicatorAgent(blockchain=blockchain)
+    token_agent = ContractAgency.get_agent(NucypherTokenAgent, registry=registry)
+    staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=registry)
+    policy_agent = ContractAgency.get_agent(PolicyManagerAgent, registry=registry)
+    adjudicator_agent = ContractAgency.get_agent(AdjudicatorAgent, registry=registry)
 
-    contract_payload = f"""
-| NuCypher Contracts |
 
-Chain .................... {blockchain.client.chain_name}
-Provider URI ............. {blockchain.provider_uri}
-Registry Path ............ {blockchain.registry.filepath}
+    try:
+        registry_filepath = registry.filepath
+    except AttributeError:
+        registry_filepath = "No filepath"
 
+    contracts = f"""
+| Contract Deployments |
 NucypherToken ............ {token_agent.contract_address}
 StakingEscrow ............ {staking_agent.contract_address}
 PolicyManager ............ {policy_agent.contract_address}
 Adjudicator .............. {adjudicator_agent.contract_address} 
     """
 
-    network_payload = f"""
-| Staking |
+    blockchain = f"""    
+| '{blockchain.client.chain_name}' Blockchain Network |
+Gas Price ................ {Web3.fromWei(blockchain.client.gas_price, 'gwei')} Gwei
+Provider URI ............. {blockchain.provider_uri}
+Registry  ................ {registry.filepath}
+    """
 
+    staking = f"""
+| Staking |
 Current Period ........... {staking_agent.get_current_period()}
 Actively Staked Tokens ... {NU.from_nunits(staking_agent.get_global_locked_tokens())}
 Published Stakes ......... {staking_agent.get_staker_population()}
-Gas Price ................ {Web3.fromWei(blockchain.client.gas_price, 'gwei')} Gwei
+Active Staking Ursulas ... {staking_agent.get_staker_population()}
```

Comment:
This prints the same thing as the line before? Rebase issue perhaps...?

Reply:
Yes, I think `published stakes` doesn't make much sense here, since there's a 1 to 1 relation between stake and stakers. I also changed the `Active Staking Ursulas` part since "Staking Ursulas" is not correct, and also `get_staker_population()` gives you all stakers, both active and inactive. Instead, I added a new functionality to get stakers partitioned by different confirmation status.

Reply:
✅